plugins/logitech-tap/fu-logitech-tap-device.c

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) 1999-2023 Logitech, Inc.
+ * Copyright (C) 2023 Richard Hughes <richard@hughsie.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include <fwupdplugin.h>
+
+#include "fu-logitech-tap-device.h"
+
+G_DEFINE_TYPE(FuLogitechTapDevice, fu_logitech_tap_device, FU_TYPE_UDEV_DEVICE)
+
+static gboolean
+fu_logitech_tap_device_write_firmware(FuDevice *device,
+				      FuFirmware *firmware,
+				      FuProgress *progress,
+				      FwupdInstallFlags flags,
+				      GError **error)
+{
+	FuLogitechTapDeviceClass *klass = FU_LOGITECH_TAP_DEVICE_GET_CLASS(device);
+	g_autofree gchar *old_firmware_version = NULL;
+	g_autoptr(GBytes) fw = NULL;
+	g_autoptr(GPtrArray) chunks = NULL;
+
+	/* save the current firmware version for troubleshooting purpose */
+	old_firmware_version = g_strdup(fu_device_get_version(device));
+
+	g_debug("update %s firmware", old_firmware_version);
+
+	/* progress */
+	fu_progress_set_id(progress, G_STRLOC);
+	fu_progress_add_step(progress, FWUPD_STATUS_DEVICE_WRITE, 100, "write");
+
+	/* get image */
+	fw = fu_firmware_get_bytes(firmware, error);
+	if (fw == NULL)
+		return FALSE;
+
+	fu_progress_set_status(progress, FWUPD_STATUS_DEVICE_WRITE);
+	chunks = fu_chunk_array_new_from_bytes(fw, 0x0, 0x0, 32);
+
+	/* write */
+	if (!klass->write_firmware(device, chunks, fu_progress_get_child(progress), error))
+		return FALSE;
+	fu_progress_step_done(progress);
+
+	/* success */
+	return TRUE;
+}
+
+static void
+fu_logitech_tap_device_set_progress(FuDevice *self, FuProgress *progress)
+{
+	fu_progress_set_id(progress, G_STRLOC);
+	fu_progress_add_flag(progress, FU_PROGRESS_FLAG_GUESSED);
+	fu_progress_add_step(progress, FWUPD_STATUS_DEVICE_RESTART, 0, "detach");
+	fu_progress_add_step(progress, FWUPD_STATUS_DEVICE_WRITE, 100, "write");
+	fu_progress_add_step(progress, FWUPD_STATUS_DEVICE_RESTART, 0, "attach");
+	fu_progress_add_step(progress, FWUPD_STATUS_DEVICE_BUSY, 0, "reload");
+}
+
+static void
+fu_logitech_tap_device_init(FuLogitechTapDevice *self)
+{
+	fu_device_add_protocol(FU_DEVICE(self), "logitech.hardware.tap");
```

Comment:
com.logitech.tap perhaps? it's reverse DNS.

Reply:
Fixed